power_grid_model_c/power_grid_model/include/power_grid_model/common/common.hpp

```diff
@@ -30,6 +30,16 @@ struct Idx2D {
     friend constexpr bool operator==(Idx2D x, Idx2D y) = default;
 };
 
+struct symmetric_t {};
+struct asymmetric_t {};
+
+template <typename T>
+concept symmetry_tag = std::derived_from<T, symmetric_t> || std::derived_from<T, asymmetric_t>;
+
+template <symmetry_tag T> constexpr bool is_symmetric = std::derived_from<T, symmetric_t>;
+
+template <symmetry_tag T> using other_symmetry_t = std::conditional_t<is_symmetric<T>, asymmetric_t, symmetric_t>;
+
```

Comment:
this is actually only used a couple of times and i don't particularly like it. If you prefer, I can move this outside `common`